tests/accesscontrol/suite.go

```diff
@@ -396,33 +396,19 @@ func testSecConRunAsNonRoot(check *checksdb.Check, env *provider.TestEnvironment
 
 	for _, put := range env.Pods {
 		check.LogInfo("Testing Pod %q in namespace %q", put.Name, put.Namespace)
-		if put.IsRunAsNonRoot() {
-			check.LogInfo("Pod %q is running as non-root", put.Name)
-			compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod is running as non-root", true))
+		nonCompliantContainers := put.GetRunAsNonRootFalseContainers(knownContainersToSkip)
+		if len(nonCompliantContainers) == 0 {
+			check.LogInfo("Pod %q is configured with RunAsNonRoot SCC parameter set to true for all of its containers", put.Name)
+			compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod is configured with RunAsNonRoot SCC parameter set to true for all of its containers", true))
 		} else {
-			check.LogError("Pod %q is running as root", put.Name)
-			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod is running as root", false))
-		}
-
-		// We are looking through both the containers and the pods separately to make compliant and non-compliant objects.
-		for _, cut := range put.Containers {
-			check.LogInfo("Testing Container %q", cut)
-			if knownContainersToSkip[cut.Name] {
-				check.LogInfo("Skipping container %q in Pod %q", cut.Name, put.Name)
-				compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(cut.Namespace, cut.Name, "Container is allowed to run as root", true))
-				continue
-			}
-
-			if cut.IsContainerRunAsNonRoot() {
-				check.LogInfo("Container %q in Pod %q is running as non-root", cut.Name, put.Name)
-				compliantObjects = append(compliantObjects, testhelper.NewPodReportObject(cut.Namespace, cut.Name, "Container is running as non-root", true))
-			} else {
-				check.LogError("Container %q in Pod %q is running as root", cut.Name, put.Name)
-				nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Container is running as root", false))
+			check.LogError("Pod %q is configured with RunAsNonRoot SCC parameter set to false for some of its containers", put.Name)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewPodReportObject(put.Namespace, put.Name, "Pod is configured with RunAsNonRoot SCC parameter set to false for some of its containers", false))
```

Comment:
Suggestion: imho, this reason and the previous log trace are not very accurate. In order to help users to debug this failing tc, I suggest adding the specific reason, which, if I'm not wrong, should be one of these three:
a) Neither of both container and pod's scc.runAsNonRoot fields were set (both are nil).
b) Container's scc.runAsNonRoot is not set (nil) but pod's scc.runAsNonRoot is explicitly set to false.
c) Container's scc.RunAsNonRoot was explicitly set to false.

Reply:
Fixed